### PR TITLE
Add explenation for registrationdeadline when user has penalty

### PR DESCRIPTION
**Pull request by amalieeh**
Fix https://github.com/webkom/lego/issues/1420

### CODE REVIEW

File: app/routes/events/components/JoinEventForm.js [hunk 1]
```diff
@@ -202,6 +210,13 @@ class JoinEventForm extends Component<Props> {
                 ? 'Åpnet '
                 : 'Åpner '}
               <Time time={event.activationTime} format="nowToTimeInWords" />
+              {this.sumPenalties() > 0 && (
```
Comment (odinuge) [inline app/routes/events/components/JoinEventForm.js]: Remember that penalties will only affect the registration time (or being added to the waiting list) if `event.heedPenalties` is set to `true`. Aka. if the event doesn't give penalties for not attending/unregistering too late, penalties will not matter. This is the case for events arranged by externals, most commonly IDI.

Comment (amalieeh) [reply app/routes/events/components/JoinEventForm.js]: I think line 207 `{!formOpen && ...` solves this?

File: app/routes/events/utils.js [hunk 2]
```diff
@@ -175,3 +175,19 @@ export const hasPaid = (chargeStatus: string) =>
   paymentSuccessMappings[chargeStatus];
 
 export const addStripeFee = (price: number) => Math.ceil(price * 1.012 + 1.8);
+
+export const sumPenalties = (penalties: Array<AddPenalty>) =>
+  sumBy(penalties, 'weight');
+
+export const penaltyHours = (penalties: Array<AddPenalty>) => {
+  switch (sumPenalties(penalties)) {
+    case 0:
```
Comment (orhanhenrik) [inline app/routes/events/utils.js]: Can remove this since default = 0

File: app/routes/events/components/JoinEventForm.js [hunk 3]
```diff
@@ -336,7 +359,15 @@ function mapStateToProps(state, { event, registration }) {
       }
     };
   }
-  return {};
+  const user = state.auth
+    ? selectUserWithGroups(state, { username: state.auth.username })
```
Comment (orhanhenrik) [inline app/routes/events/components/JoinEventForm.js]: You can use `selectUserByUsername` instead, since groups aren't used

**Comment by orhanhenrik** [comment]
What's the status on this @amalieeh? 👍

**Comment by orhanhenrik** [comment]
Is this dead? almost 1 year since last push 😭

File: app/routes/events/utils.js [hunk 4]
```diff
@@ -189,3 +189,21 @@ export const registrationCloseTime = (event: Event) =>
...
+      return 0;
+    case 1:
+      return 3;
+    case 2:
+      return 12;
+    case 3:
+      return 1337;
```
Comment (mrfarstad) [inline app/routes/events/utils.js:205]: 🌮 🎉

**Pull request merged**